identity-service/src/txRelay.js

```diff
@@ -92,59 +91,38 @@ const sendTransactionInternal = async (req, web3, txProps, reqBodySHA) => {
   let redisLogParams
   let wallet = selectWallet()
 
+  // If all wallets are currently in use, keep iterating until a wallet is freed up
   while (!wallet) {
     await delay(200)
     wallet = selectWallet()
   }
 
   try {
-    req.logger.info('relayTx - selected wallet', wallet.publicKey)
-    const privateKeyBuffer = Buffer.from(wallet.privateKey, 'hex')
-    const walletAddress = EthereumWallet.fromPrivateKey(privateKeyBuffer)
-    const address = walletAddress.getAddressString()
-    if (address !== wallet.publicKey.toLowerCase()) {
-      throw new Error('Invalid relayerPublicKey')
-    }
-
-    const gasPrice = await getGasPrice(req, web3)
-    const nonce = await web3.eth.getTransactionCount(address)
-
-    const txParams = {
-      nonce: web3.utils.toHex(nonce),
-      gasPrice: gasPrice,
-      gasLimit: gasLimit ? web3.utils.numberToHex(gasLimit) : DEFAULT_GAS_LIMIT,
-      to: contractAddress,
-      data: encodedABI,
-      value: '0x00'
-    }
-
-    const tx = new EthereumTx(txParams)
-    tx.sign(privateKeyBuffer)
-
-    const signedTx = '0x' + tx.serialize().toString('hex')
+    logger.info('relayTx - selected wallet', wallet.publicKey)
+    const resp = await createAndSendTransaction(wallet, contractAddress, '0x00', web3, gasLimit, encodedABI)
```

Comment:
you should destructure to { receipt and txParams } here. i was confused for a sec thinking the tx receipt had a txParams property haha